deepchem/feat/smiles_featurizers.py

```diff
@@ -0,0 +1,227 @@
+"""
+Featurizer implementations used in ChemCeption and Smiles2Vec models.
+"""
+
+from __future__ import division
+from __future__ import unicode_literals
+
+__author__ = "Vignesh Ram Somnath"
+__license__ = "MIT"
+
+import numpy as np
+import pandas as pd
+from deepchem.feat import Featurizer
+
+PAD_TOKEN = "<pad>"
+OUT_OF_VOCAB_TOKEN = "<unk>"
+
+
+def create_char_to_idx(filename,
+                       max_len=250,
+                       smiles_field="smiles",
+                       verbose=False):
+  """Creates a dictionary with character to index mapping.
+
+    Parameters
+    ----------
+    filename: str,
+        Name of the file containing the SMILES strings
+    max_len: int, default 250
+        Maximum allowed length of the SMILES string
+    smiles_field: str, default smiles
+        Field indicating the SMILES strings int the file.
+    verbose: bool, default True
+        Whether to print the progress
+    """
+  smiles_df = pd.read_csv(filename)
+  char_set = set()
+  for smile in smiles_df[smiles_field]:
+    if len(smile) <= max_len:
+      char_set.update(set(smile))
+
+  unique_char_list = list(char_set)
+  unique_char_list += [PAD_TOKEN, OUT_OF_VOCAB_TOKEN]
+  if verbose:
+    print("Number of unique characters: ", len(unique_char_list))
+
+  char_to_idx = {letter: idx for idx, letter in enumerate(unique_char_list)}
+
+  if verbose:
+    print(unique_char_list)
+  return char_to_idx
+
+
+class SmilesToSeq(Featurizer):
+
+  def __init__(self, char_to_idx, max_len=250, pad_len=10, **kwargs):
+    """
```

Comment:
Can you add a paragraph or two to the docstring to explain what the class does and how to use it?

Reply:
Sure, will do!